bee-message/src/parents.rs

```diff
@@ -51,19 +144,106 @@ impl Packable for Parents {
     }
 
     fn unpack<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Self::Error> {
-        let parents_len = u8::unpack(reader)? as usize;
+        let _parents_len = u8::unpack(reader)? as usize;
+
+        // const MESSAGE_PARENTS_RANGE: RangeInclusive<usize> = 1..=8;
+
+        // if !MESSAGE_PARENTS_RANGE.contains(&parents_len) {
```

Comment:
The check still needs to happen as it is part of the protocol. If there is not check, an invalid message could be accepted by bee but denied by hornet, leading to a consensus diff.

Reply:
you're absolutely right. Preventing the check as a whole is impossible as long as it's serialized somewhere in `Vec` "manner". 
Tbh, I don't care so much for this PR to be merged. I see this PR more like an interesting case study to have a conversation about. I think the proposed `Parents` wrapper is something we should merge, though. We can use it to have the runtime checks in there.